cyclicLinkedList2.java

```diff
@@ -0,0 +1,30 @@
+// Did this code successfully run on Leetcode : Yes
+// Time Complexity :
+// Space Complexity :O(1)
+
+public class Solution {
+    public ListNode detectCycle(ListNode head) {
+        ListNode slow=head;
+        ListNode fast = head;
+        boolean isCycle = false;
+        if(head==null) return null;
+        while(fast!=null && fast.next!=null ){
+            slow=slow.next;
+            fast=fast.next.next;
+            if(slow==fast){
+                isCycle = true;
+                break;
+            } 
+        }
+        if(isCycle){
+            ListNode slow2 =head;
```

Comment:
Try to re-use the fast variable instead of creating new variables. It doesn't make any difference in the space complexity but it does make a point that you are trying to minimize the space